### ISSUE TITLE
Unsuitably suggests making initialize a predicate

### ISSUE THREAD

**Issue by forthrin** [rubocop/rubocop#14493]
## Steps to reproduce the problem

```ruby
# Foo
class Foo < Set
  def initialize
    super
    add?('Default')
  end
end

pp Foo.new
```

## Actual behavior

```Predicate method names should end with `?`. [Naming/PredicateMethod]```

## Suggested behavior

```Boolean test void. `initialize` always returns object. [Lint/InitializeVoidTest]```

## RuboCop version

```
ruby      2025-08-01 30a20bc16
rubocop   2025-08-12 c34098ee0
System Version: macOS 15.6 (24G84)
Model Identifier: MacBookAir10,1
```

**Comment by koic** [comment]
Since this `add?('Default')` appears to be logic without meaning, it may be suggesting that some predicate method name should originally be defined, and therefore it cannot be definitively judged at this point to be incorrect behavior. Can you explain the intention or use case for calling the predicate method at the end of the `initialize` method?

**Comment by forthrin** [comment]
In the original code, `add?` overrides Set:add? with some extra functionality. The class has no `add`, because this is never used anywhere in the code. Thus `add?` is used in the initialize though we are not interested in the result there.

Renaming `initialize` to `initialize?`, which the cop naively hints at, obviously makes no sense. So best to just silence the warning for all `initialize` functions in classes seems appropriate.

**Comment by koic** [comment]
Hm. In this case, using `Set#add` rather than `Set#add?` appears appropriate. This would eliminate problems even in the `initialize` method definition. If using a predicate method like `Set#add?` is actually correct, there is a risk of overlooking cases where some other method ending with `?` (rather than `initialize?`) is what is truly expected.

**Comment by forthrin** [comment]
Yes, the class could implement an `add` function for functions that don't need the result of the test, but want the rest of the functionality.

However initialize is unique in that it always returns the newly created object, whereas all other functions return the last statement. Therefore the cop does not make sense for `AnyClass:initialize`. Again, `AnyClass.initialize?` does not exist as a concept. An alternative could be to hint about `Boolean result in void context` for programmers that for a momentary brain lapse think they can get a boolean result back from a constructor. But I lean towards just silencing.

**Comment by koic** [comment]
OK. Since it is unlikely that a user would mistake the definition of the `initialize` method for that of another method, methods returning a value within `initialize` should not be inspected. I've opened #14495 to address this issue.
